packages/merge-styles/src/mergeStyleSets.ts

```diff
@@ -223,12 +223,17 @@ export function mergeCssSets(
       const styles: IStyle = (concatenatedStyleSet as any)[styleSetArea];
 
       const { classes, objects } = extractStyleParts(styles);
-      const registration = styleToRegistration(options || {}, { displayName: styleSetArea }, objects);
 
-      registrations.push(registration);
+      if (objects && objects.length === 0) {
```

Comment:
If `object` is falsey, wouldn't you also want to fall back to the optimization?

Maybe adjust the logic:

```tsx
if (objects?.length) { // if objects is truthy and has a length
  // do normal stuff
} else {
  // do new stuff 
}